src/assets/Generator.Shared/NextLinkOperationImplementation.cs

```diff
@@ -205,15 +205,18 @@ internal static bool TryGetApiVersion(Uri startRequestUri, out ReadOnlySpan<char
                 return null;
             }
 
-            // Handle final-state-via options: https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-long-running-operation-options
-            switch (_finalStateVia)
+            if (_requestMethod == RequestMethod.Post)
             {
-                case OperationFinalStateVia.Location when _originalResponseHasLocation:
-                    return _lastKnownLocation;
-                case OperationFinalStateVia.OperationLocation or OperationFinalStateVia.AzureAsyncOperation:
-                    return null;
-                case OperationFinalStateVia.OriginalUri:
-                    return _startRequestUri.AbsoluteUri;
+                // Handle final-state-via options: https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-long-running-operation-options
+                switch (_finalStateVia)
+                {
+                    case OperationFinalStateVia.Location when _originalResponseHasLocation:
+                        return _lastKnownLocation;
+                    case OperationFinalStateVia.OperationLocation or OperationFinalStateVia.AzureAsyncOperation:
+                        return null;
+                    case OperationFinalStateVia.OriginalUri:
+                        return _startRequestUri.AbsoluteUri;
+                }
```

Comment:
I realize @lmazuel said that `final-state-via` was designed for `POST`, but I'm not ready to assume that everyone followed the spec. We've already seen so many variations of LROs across services we really need to consider *what is* and not *what should be*. Even the current REST API standards for LROs are implemented by very few services since we changed them recently...partly to consolidate.

@johanste what do you think?

Reply:
Talking from Python side, we started to have final-state-via filtered to POST only. We attempted to generalize it to any verb, which was a disaster and broke the CLI, because service teams misunderstood the option (they thought it was to guide the flow, while this is an option to decide if a final GET is needed or not).

So I know by unfortunate experience there is Swagger out there which are false, and mis-use the option compared to what the service does.

We ended up do a hot fix of `azure-core` and add back the POST test:
https://github.com/Azure/azure-sdk-for-python/blob/cb2c5b7745f211783450477885be353642dfb1c3/sdk/core/azure-core/azure/core/polling/base_polling.py#L230-L243

https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/core/azure-core/CHANGELOG.md#bugs-fixed-6

Reply:
One more reference: https://github.com/Azure/azure-sdk-for-python/pull/22989/files#diff-d0de953d24f7412fe6b9eae87972a0fb383d7d4d865cad497c872345390f2bb3R240